docs/snippets/ov_extensions.cpp

```diff
@@ -2,20 +2,124 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 #include <openvino/openvino.hpp>
-#include <openvino/core/op_extension.hpp>
+//! [add_extension_header]
+//#include <openvino/core/op_extension.hpp>
+//! [add_extension_header]
+//! [add_frontend_extension_header]
+#include <openvino/frontend/extension.hpp>
+//! [add_frontend_extension_header]
+
+//! [frontend_extension_Identity_header]
+#include <openvino/frontend/extension.hpp>
+//! [frontend_extension_Identity_header]
+
+//! [frontend_extension_ThresholdedReLU_header]
+#include <openvino/opsets/opset8.hpp>
+//! [frontend_extension_ThresholdedReLU_header]
+
 #include <identity.hpp>
 
+
 int main() {
 {
 //! [add_extension]
 ov::Core core;
-// Use operation type to add operation extension 
+
+// Use operation type to add operation extension
 core.add_extension<TemplateExtension::Identity>();
-// or you can add operation extension to this method
+
+// or you can add operation extension object which is equivalent form
 core.add_extension(ov::OpExtension<TemplateExtension::Identity>());
 //! [add_extension]
 }
 {
+ov::Core core;
+
+//! [add_frontend_extension]
+// Register mapping for new frontends: FW's "TemplateIdentity" operation to TemplateExtension::Identity
+core.add_extension(ov::frontend::OpExtension<TemplateExtension::Identity>("Identity"));
+
+// Register more sophisticated mapping with decomposition
+core.add_extension(ov::frontend::ConversionExtension(
+    "Identity",
+    [](const ov::frontend::NodeContext& context) {
+        // Arbitrary decomposition code here
+        // Return a vector of operation outputs
+        return ov::OutputVector{ std::make_shared<TemplateExtension::Identity>(context.get_input(0)) };
+    }));
+//! [add_frontend_extension]
+}
+{
+//! [frontend_extension_Identity]
+auto extension1 = ov::frontend::OpExtension<TemplateExtension::Identity>("Identity");
+
+// or even simpler if original FW type and OV type of operations match, that is "Identity"
+auto extension2 = ov::frontend::OpExtension<TemplateExtension::Identity>();
+//! [frontend_extension_Identity]
+
+//! [frontend_extension_read_model]
+ov::Core core;
+// Add arbitrary number of extensions before calling read_model method
+core.add_extension(ov::frontend::OpExtension<TemplateExtension::Identity>());
+core.read_model("/path/to/model.onnx");
+//! [frontend_extension_read_model]
+
+//! [frontend_extension_MyRelu]
+core.add_extension(ov::frontend::OpExtension<>("Relu", "MyRelu"));
+//! [frontend_extension_MyRelu]
+
+//! [frontend_extension_CustomOperation]
+class CustomOperation : public ov::op::Op {
+
+    std::string attr1;
+    int attr2;
+
+public:
+
+    OPENVINO_OP("CustomOperation");
```

Comment:
to fix issue #82704 need to move `class CustomOperation` declaration out of `main()` scope

Reply:
thanks, fixed